src/core/handlers/RequestHandler.ts

```diff
@@ -184,6 +186,36 @@ export abstract class RequestHandler<
     })
   }
 
+  private static cookiesForRequestCache = new WeakMap<
+    Request,
+    Record<string, string>
+  >()
+  protected parseAllRequestCookiesOrGetFromCache(
+    request: Request,
+  ): Record<string, string> {
+    let cookies = RequestHandler.cookiesForRequestCache.get(request)
+    if (!cookies) {
+      cookies = getAllRequestCookies(request)
+      RequestHandler.cookiesForRequestCache.set(request, cookies)
+    }
+    return cookies
+  }
+
+  private static matchRequestUrlCache = new Map<string, Match>()
```

Comment:
Why not `WeakMap` here also? Are we planning on freeing these resources manually?

Reply:
WeakMap can't accept string keys, but mostly because this cache is actually more valuable if it persists between requests.  Since multiple requests can have the same URL string, this maintains the cache across requests, avoiding the new URL overhead for multiple requests.

It's a slight extra optimization, but these shouldn't be so huge where they become an issue since it should be relatively small

We could later consider something more like an lru here if we need to evict things

Reply:
Got it. The choice of `Map` is clear.

Reply:
My main concern is that these changes wouldn't hurt the readability of the code. The benefit they provide is tangible but rather minimal. I'm fine with having them as long as the code doesn't become a jungle. I suggested a `cached` helper to make this common cached logic reused, I'd like to know what you think about it.